[WPE][GTK] run-test-runner: add support for WPE and GTK ports

obyknovenius opened the pull request:
#### 93f08f00af33cfe2cdd7acd8b984947b8b2396ac
<pre>
[WPE][GTK] run-test-runner: add support for WPE and GTK ports
<a href="https://bugs.webkit.org/show_bug.cgi?id=270903">https://bugs.webkit.org/show_bug.cgi?id=270903</a>

Reviewed by Adrian Perez de Castro.

This patch adds `runUnixWebKitApp()` subroutine which configures the
environment and runs an executable in the build directory.
It is used to run `WebKitTestRunner` on WPE and GTK.

* Tools/Scripts/webkitdirs.pm:
(runUnixWebKitApp):
(runWebKitTestRunner):

Canonical link: <a href="https://commits.webkit.org/276078@main">https://commits.webkit.org/276078@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/79dac50a92da9c2246076fba7e047fd4579f62c5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43573 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22614 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45998 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46214 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39702 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45877 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26432 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20029 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36008 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44147 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19635 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37518 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16981 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17181 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38580 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1629 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/37036 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39721 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38885 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47768 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/43206 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18610 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15207 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42809 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20030 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41491 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9717 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20210 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/50208 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19661 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10119 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
Just a minor comment to fix before landing, otherwise LGTM.

aperezdc commented on Tools/Scripts/webkitdirs.pm:
```diff
@@ -3320,6 +3320,18 @@ sub runMacWebKitApp($;$)
+{
+    my ($appPath) = @_;
+    my $productDir = productDir();
+    print "Starting @{[basename($appPath)]} with build WebKit in $productDir.\n";
```
Typo: `s/build/built/`

obyknovenius replied:
Thanks. Fixed.

obyknovenius committed:
[WPE][GTK] run-test-runner: add support for WPE and GTK ports
https://bugs.webkit.org/show_bug.cgi?id=270903

Reviewed by Adrian Perez de Castro.

This patch adds `runUnixWebKitApp()` subroutine which configures the
environment and runs an executable in the build directory.
It is used to run `WebKitTestRunner` on WPE and GTK.

* Tools/Scripts/webkitdirs.pm:
(runUnixWebKitApp):
(runWebKitTestRunner):

Canonical link: https://commits.webkit.org/276078@main
